src/libopensc/card-openpgp.c

```diff
@@ -956,6 +960,83 @@ pgp_get_card_features(sc_card_t *card)
 			}
 		}
 
+#ifdef ENABLE_OPENSSL
+		if (priv->ext_caps & EXT_CAP_KDF_DO) {
+			pgp_blob_t *kdf_blob = NULL;
+			if ((pgp_get_blob(card, priv->mf, DO_KDF, &kdf_blob) == SC_SUCCESS) && (kdf_blob->len >= 3)) {
+				uint8_t *data = kdf_blob->data;
+				priv->pin_kdf_info = NULL;
+				/* KDF_ITERSALTED_S2K */
+				size_t i = 0;
+				pgp_pin_kdf_info *pin_kdf_info = (pgp_pin_kdf_info *)calloc(1, sizeof(pgp_pin_kdf_info));
+				if (!priv) {
+					free(priv);
+					pgp_free_blob(kdf_blob);
```

Comment:
I think this does not need to be freed as it will still be "cached" in the priv data